src/CMakeLists.txt

```diff
@@ -273,18 +273,19 @@ if(CMAKE_CROSSCOMPILING)
     target_sources(${bootloader}.elf PRIVATE ${PLATFORM-BITBOX02-SOURCES})
   endforeach(bootloader)
 
-  target_compile_definitions(bootloader-development.elf PRIVATE BOOTLOADER_DEVDEVICE)
-  target_compile_definitions(bootloader-development-locked.elf PRIVATE BOOTLOADER_DEVDEVICE BOOTLOADER_PRODUCTION)
+  target_compile_definitions(bootloader.elf PRIVATE PRODUCT_BITBOX_MULTI)
+  target_compile_definitions(bootloader-development.elf PRIVATE PRODUCT_BITBOX_MULTI BOOTLOADER_DEVDEVICE)
```

Comment:
Also `BITBOX02` instead of `BITBOX` :)

Reply:
The idea was that we only need one macro on the command line. Which could be `PRODUCT_`. All other macros would be defined in config files.

Reply:
Another idea I had right now would be to mangle the target name and put that on the CLI. So every single target has to be mentioned in the config file. For example `-DBOOTLOADER` would be the `bootloader.bin` target. And `-DBOOTLOADER_BASE_PRODUCTION` would be the base production bootlaoder. Those names could be automatically generated and then we would collect all defines in a `config.h` file instead.

Reply:
It will mean that the config file will look a bit repetitive. But it could also be less error prone

Reply:
You wouldn't have to do `make VERBOSE=1` to verify that all the command lines are correct while compiling.

Reply:
Ah, and another thing is that since we will migrate to "always define all macros" you would have to define everything that is "not in use" also. So the btconly would look like `-DAPP_U2F=0 -DAPP_ETH=0...` and this will of course apply to every target with every single option. Much easier to handle in a config file IMO.

Reply:
+1 for the config file. `#define`ing all macros in each Makefile is just too much of a pain for the few configurations that we have.

To address @benma's comment, I would point out that `PLATFORM` refers to different circuitry, while `EDITION` is just different software of the same electronics. So all firmwares running on the same `PLATFORM` have access to the same hw. I would ditch the `PRODUCT` and stick to`PLATFORM` and `EDITION` (or change it to `PRODUCT`+`EDITION` or whatever if it's clearer), unless I'm missing something of course.

Also: this needs urgent merging, so happy to leave this as it is and discuss this offline later on.

Reply:
We can define `PLATFORM_` and `EDITION_` on the command line. But then you are not allowed to use those defines anywhere except in the config file. Since we want to avoid `defined` in `#if defined(...)`.

It would look like this: `-DPLATFORM_BITBOX02 -DEDITION_BTCONLY`.

I suggest `-DPRODUCT_BITBOX02_BTCONLY` and then define `PLATFORM_BITBOX02` and `EDITION_BTCONLY` in the config file so that we can use them in other files.

What macros do you suggest in the config file then?

If you meant that we should have: `-DPLATFORM_BITBOX02=1 -DPLATFORM_BITBOXBASE=0 -DBOOTLOADER=0 -DEDITION_BTCONLY=1 -DEDITION_MULTI=0`. Then I'm strongly against that. Because I want to move the configuration away from the build system and into a header file.

Reply:
My idea would be to define (from the commandline/Makefile) `-DPLATFORM=BITBOX02 -DEDITION=BTCONLY`, `-DPLATFORM=BITBOXBASE`, etc.; the config file can parse these macros and generate the relevant `#define`s. The same config file can also warn if a `#define`d value is invalid. That is, I'm strongly supporting defines-as-enums here.